### PR TITLE
Fixed `Extending Filtering` documentation example

### LINKED ISSUE #3442: Your documentation page has non compiling code
I was trying to implement case insensitive filtering and found this example on you [docs page](https://chillicream.com/docs/hotchocolate/api-reference/extending-filtering#extending-iqueryable). Bit it gives compile errors.
AddProvider() method does not exists. 

```
public class CustomFilteringConvention : FilterConvention
{
    protected override void Configure(IFilterConventionDescriptor descriptor)
    {
        descriptor.AddDefaults();
        descriptor.AddProvider(
            new QueryableFilterProvider(
                x => x
                    .AddDefaultHandlers()
                    .AddFieldHandler<QueryableStringInvariantEqualsHandler>()));
    }
}
```

According to [issue reports](https://github.com/ChilliCream/hotchocolate/issues/2687) there is another code snippet. So might this should be used:

```
  public class CustomFilteringConvention : FilterConvention
  {
      protected override void Configure(IFilterConventionDescriptor descriptor)
      {
          descriptor.AddDefaults();
          descriptor.Operation(DefaultFilterOperations.Equals).Name("equals");
          descriptor.AddProviderExtension(
              new QueryableFilterProviderExtension(
                  x => x.AddFieldHandler<QueryableStringInvariantEqualsHandler>()));
      }
    }
```

### PR DESCRIPTION
**Pull request by tobias-tengler**
Closes #3442

### CODE REVIEW

File: website/src/docs/hotchocolate/api-reference/extending-filtering.md [hunk 1]
```diff
@@ -402,13 +402,12 @@ services.AddGraphQLServer()
 ```
 
 To make this registration easier, HotChocolate also supports convention and provider extensions.
-Instead of creating a customer `FilterConvention`, you can also do the follwing:
+Instead of creating a custom `FilterConvention`, you can also do the follwing:
 
 ```csharp
 services.AddGraphQLServer()
-    .AddFiltering<CustomFilteringConvention>();
     .AddConvention<IFilterConvention>(
```
Comment (PascalSenn) [inline website/src/docs/hotchocolate/api-reference/extending-filtering.md:409]:
```suggestion
    .AddFiltering()
    .AddConvention<IFilterConvention>(
```

File: website/src/docs/hotchocolate/api-reference/extending-filtering.md [hunk 2]
```diff
@@ -402,13 +402,12 @@ services.AddGraphQLServer()
...
-        new FilterConventionConvention(
+        new FilterConvention(
```
Comment (PascalSenn) [inline website/src/docs/hotchocolate/api-reference/extending-filtering.md:410]:
```suggestion
        new FilterConventionExtension(
```

**Review by PascalSenn — CHANGES_REQUESTED** [review]
Thanks for the PR @tobias-tengler 
I think this should do

**Comment by tobias-tengler** [comment]
@PascalSenn this should be ready to go. I'm not sure why your requested changes are still blocking the PR, I already resolved them...

**Review by PascalSenn — APPROVED** [review]

**Pull request merged**